File: backend/app/http/biz/handler/file/file_api.go
package file

import (
	"context"

	"backend/app/http/biz/service"
	"backend/app/http/biz/utils"
	file "backend/app/http/hertz_gen/file"
	"github.com/cloudwego/hertz/pkg/app"
	"github.com/cloudwego/hertz/pkg/protocol/consts"
)

// NewMultiUpload .
// @router /v1/auth/file/multi_upload [POST]
func NewMultiUpload(ctx context.Context, c *app.RequestContext) {
	var err error
	var req file.NewMultiUploadReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &file.NewMultiUploadResp{}
	resp, err = service.NewNewMultiUploadService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// GetMultiUploadUri .
// @router /v1/auth/file/multi_upload_uri [POST]
func GetMultiUploadUri(ctx context.Context, c *app.RequestContext) {
	var err error
	var req file.GetMultiUploadUriReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp, err := service.NewGetMultiUploadUriService(ctx, c).Run(&req)

	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}
	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// CompleteMultipart .
// @router /v1/auth/file/complete_multipart [POST]
func CompleteMultipart(ctx context.Context, c *app.RequestContext) {
	var err error
	var req file.CompleteMultipartReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp, err := service.NewCompleteMultipartService(ctx, c).Run(&req)

	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}
	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// GetSuccessChunks .
// @router /v1/auth/file/success_chunks [POST]
func GetSuccessChunks(ctx context.Context, c *app.RequestContext) {
	var err error
	var req file.GetSuccessChunksReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp, err := service.NewGetSuccessChunksService(ctx, c).Run(&req)

	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}
	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

File: frontend/src/api/fileApi.ts
// @ts-ignore
/* eslint-disable */
import request from '@/utility/request'

/** 完成分块上传 POST /v1/auth/file/complete_multipart */
export async function completeMultipart(body: API.CompleteMultipartReq, options?: { [key: string]: any }) {
    return request<API.CompleteMultipartResp>('/v1/auth/file/complete_multipart', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}

/** 上传文件 POST /v1/auth/file/multi_upload */
export async function newMultiUpload(body: API.NewMultiUploadReq, options?: { [key: string]: any }) {
    return request<API.NewMultiUploadResp>('/v1/auth/file/multi_upload', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}

/** 获取大文件上传请求的 URI POST /v1/auth/file/multi_upload_uri */
export async function getMultiUploadUri(body: API.GetMultiUploadUriReq, options?: { [key: string]: any }) {
    return request<API.GetMultiUploadUriResp>('/v1/auth/file/multi_upload_uri', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}

/** 获取已经上传的分块 POST /v1/auth/file/success_chunks */
export async function getSuccessChunks(body: API.GetSuccessChunksReq, options?: { [key: string]: any }) {
    return request<API.GetSuccessChunksResp>('/v1/auth/file/success_chunks', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}
